### PR TITLE
Add initial Colormap implementation with keypoints

### PR DESCRIPTION
**Pull request by kitizz**
- Introduce `Colormap` struct to map parameters to colors.
- Support uniform and non-uniform keypoints map inputs.
- Support for sRBG color interpolation for now.
- Add unit tests:
    - Uniform and non-uniform keypoints.
    - Edge cases (empty colormap, single color).
    - Out-of-bounds parameter handling.

---

* Related to #188
* [x] I have followed the instructions in the PR template

### CODE REVIEW

File: egui_plot/src/colormap.rs [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+/// Currently, colors are interpolated from provided keypoints in sRGB space.
+/// Keypoints may be uniformly distributed ([`Colormap::new_uniform`])
+/// or have explicit positions ([`Colormap::new_with_positions`]).
+///
+/// If building your own Colormap with explicit positions, it is strongly recommended that
+/// keypoints cover the full range from 0.0 to 1.0. This is because PlotItem consumers
+/// make this assumption and will only sample the colormap in this range.
+/// However, this is not enforced, so feel free to go against this for your own special use-cases.
+pub struct Colormap {
+    data: ColormapData,
```
Comment (kitizz) [inline egui_plot/src/colormap.rs]: I'm anticipating additional fields here like the interpolation function to use.

File: egui_plot/src/colormap.rs [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    /// but be aware that PlotItems will only consider the [0.0, 1.0] range.
+    pub fn new_with_positions(keypoints: Vec<(f32, Color32)>) -> Self {
+        Self {
+            data: ColormapData::new_nonuniform(keypoints),
+        }
+    }
+
+    /// Get color at a given position.
+    pub fn get(&self, t: f32) -> Color32 {
+        self.data.get(t, ColorInterpolation::SRGB)
```
Comment (kitizz) [inline egui_plot/src/colormap.rs]: I haven't exposed this yet because I'm wondering if it will be more efficient to store the colormap colors in the desired interpolation space. And then convert them after interpolation. In this case, the interpolation function is the same for everything, it's just a lerp. And then the functions would be the conversions to/from sRGB.
